File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def numProd(num: int) -> int:
            prod = 1
            while num > 0:
                prod *= (num % 10)
                num //= 10
            return prod
        for i in range(n, n + 10):
            if numProd(i) % t == 0:
                return i
        return 0

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        for(int i = n; i < n + 10; i++)
        {
            if(numProd(i) % t == 0)
            {
                return i;
            }
        }
        return 0;
    }
private:
    int numProd(int num)
    {
        int prod = 1;
        while(num > 0)
        {
            prod *= (num % 10);
            num /= 10;
        }
        return prod;
    }
};